File: json_encode/json_encode.java
import com.google.gson.Gson;

import java.util.HashMap;

public class json_encode {
    // ATOMIC_FUNCTION:  json_encode(Object x)
    // TAGS: json
    // LANG:  java
    // FUNCTION: json_encode(Object x)
    public String json_encode(Object x){
        Gson gson = new Gson();
        return  gson.toJson(x);
    }
    public static void main(String[] args){
        HashMap<String, Integer> start_val= new HashMap<>();
        start_val.put("atomic",1);
        json_encode j = new json_encode();
        String expected_end_val="{\"atomic\":1}";
        String end_val = j.json_encode(start_val);

        // TEST
        if(end_val.equals(expected_end_val)) {
            System.out.println("PASS");
        } else {

            System.out.println("FAIL");
        }
    }

}

File: json_encode/json_encode.js
// ATOMIC_FUNCTION: json_encode(x) 
// TAGS: json_encode
// LANG:  node.js
// FUNCTION: json_encode(x) 
function json_encode(x) {
    return JSON.stringify(x);
}
var start_val = { x: 5 };
var expected_end_val = '{"x":5}';
var end_val = json_encode(start_val);

// TEST
if(end_val==expected_end_val){
    console.log('PASS');
}else{
	
    console.log('FAIL');
}
